### PR TITLE
GH Actions refinements, AZP removal

### PR DESCRIPTION
**Pull request by michdolan**
This PR removes AZP from OCIO CI, and is a refinement pass to GH Actions:

- Split SonarCloud and "latest" builds into their own workflow. They will no longer show up in PR checks, and will have their own build status badge.
- Added a GHA status badge per workflow to README.md
- Add VFX Reference Platform CY2020 to the build. Two CY2019 jobs are preserved for testing Python 2.7 (there aren't many other significant differences in terms of OCIO dependencies)
- Step and comment cleanup

I wanted to point out the contents of one comment in particular, which references a nice upgrade that GHA adds to our CI:

> GH Actions (Free plan) supports 20 concurrent jobs, with 5 concurrent macOS jobs. This workflow tries to utilize (but not exceed) that budget to promote timely CI.

Signed-off-by: Michael Dolan <michdolan@gmail.com>

### CODE REVIEW

**Comment by michdolan** [comment]
I renamed the workflow and jobs slightly, which will likely result in this PR saying it is waiting for required checks, but the new named builds are present too. I won't be able to change the required checks until after this is merged, so this checks state won't be fixable.

File: .github/workflows/analysis_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.#
```
Comment (hodoulp) [inline .github/workflows/analysis_workflow.yml:2]: Is it needed to have a # at the end of the line?

File: .github/workflows/analysis_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                -DCMAKE_CXX_FLAGS="-g -O0 -fprofile-arcs -ftest-coverage" \
+                -DCMAKE_CXX_OUTPUT_EXTENSION_REPLACE=ON \
+                -DCMAKE_EXE_LINKER_FLAGS="-lgcov" \
+                -DOCIO_BUILD_DOCS=ON \
+                -DOCIO_BUILD_GPU_TESTS=OFF \
+                -DOCIO_INSTALL_EXT_PACKAGES=ALL \
+                -DOCIO_WARNING_AS_ERROR=ON \
+                -DPYTHON_EXECUTABLE=$(which python)
+        working-directory: _build
+      - name: Build OCIO with build-wrapperr
```
Comment (hodoulp) [inline .github/workflows/analysis_workflow.yml:185]: `build-wrapperr` with 2 'r' at the end!

File: .github/workflows/analysis_workflow.yml (hunk 1 above)
Comment (michdolan) [reply .github/workflows/analysis_workflow.yml]: Fixed.

File: .github/workflows/analysis_workflow.yml (hunk 2 above)
Comment (michdolan) [reply .github/workflows/analysis_workflow.yml]: Good catch. Fixed.

**Pull request merged**